packages/@aws-cdk/aws-ec2/lib/vpc-ref.ts

```diff
@@ -335,3 +361,34 @@ export interface VpcSubnetImportProps {
    */
   subnetId: string;
 }
+
+/**
+ * If the placement strategy is completely "default", reify the defaults so
+ * consuming code doesn't have to reimplement the same analysis every time.
+ *
+ * Returns "private subnets" by default.
+ */
+function reifyDefaultsPlacement(placement: VpcPlacementStrategy): VpcPlacementStrategy {
+    if (placement.subnetsToUse !== undefined && placement.subnetName !== undefined) {
+      throw new Error('At most one of subnetsToUse and subnetName can be supplied');
+    }
+
+    if (placement.subnetsToUse === undefined && placement.subnetName === undefined) {
+      return { subnetsToUse: SubnetType.Private };
+    }
+
+    return placement;
+}
+
+/**
+ * Describe the given placement strategy
+ */
+function describePlacement(placement: VpcPlacementStrategy) {
+  if (placement.subnetsToUse !== undefined) {
+    return `'${DEFAULT_SUBNET_NAME[placement.subnetsToUse]}' subnets`;
+  }
+  if (placement.subnetName !== undefined) {
+    return `subnets named '${placement.subnetName}'`;
+  }
+  return '???';
```

Comment:
If you get there you've effectively reached the default behavior, I believe. Then you should probably return `private subnets`. If you are willing to assume that the defaults have always been reified before you get here, then why not throw instead of returning `???`?

Reply:
I was thinking to throw, but at the same time I didn't want to block a customer from using this construct if we ever forget a code path. Rather have ugly behavior but allow the user to continue (so that we can fix the issue when we have capacity) rather than block them completely by throwing an error and now we have to scramble to get a fix out.

Reply:
Right but then at least avoid the question marks... Maybe `JSON.stringify` what you've been given.